com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -83,23 +89,24 @@ protected String getStepErrorMessage(DelegateExecution context) {
         return Messages.ERROR_DELETING_SERVICES;
     }
 
-    private Map<String, CloudServiceExtended> getServicesData(List<String> serviceNames, ExecutionWrapper execution) {
+    private List<CloudServiceExtended> getServicesData(List<String> serviceNames, ExecutionWrapper execution) {
         CloudControllerClient client = execution.getControllerClient();
 
         return serviceNames.parallelStream()
-            .map(name -> client.getService(name, false))
-            .filter(Objects::nonNull)
-            .map(service -> ImmutableCloudServiceExtended.builder()
-                .metadata(service.getMetadata())
-                .name(service.getName())
-                .build())
-            .collect(Collectors.toMap(CloudEntity::getName, e -> e));
+                           .map(name -> client.getService(name, false))
+                           .filter(Objects::nonNull)
+                           .map(service -> ImmutableCloudServiceExtended.builder()

```

Comment:
Move this mapping to a method and pass it as a method reference to .map()

Reply:
This actually is not possible because of need to pass CloudControllerClient